java/src/jmri/jmrix/mqtt/MqttConsistManager.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Consist Manager for use with the MqttConsist class for the
+ * consists it builds.
+ *
+ * @author Dean Cording Copyright (C) 2023
+ */
+package jmri.jmrix.mqtt;
+
+import jmri.Consist;
+import jmri.LocoAddress;
+import jmri.DccLocoAddress;
+import jmri.implementation.AbstractConsistManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import javax.annotation.Nonnull;
+
+public class MqttConsistManager extends AbstractConsistManager {
+
+    protected MqttSystemConnectionMemo adapterMemo;
+
+    /**
+     * Constructor - call the constructor for the superclass, and initialize the
+     * consist reader thread, which retrieves consist information from the
+     * command station.
+     *
+     * @param memo the associated connection memo
+     */
+    public MqttConsistManager(MqttSystemConnectionMemo memo) {
+        super();
+        adapterMemo = memo;
+    }
+
+    public void setSendTopic(@Nonnull String sendTopicPrefix) {
+        this.sendTopicPrefix = sendTopicPrefix;
+    }
+
+    @Nonnull
+    public String sendTopicPrefix = "cab/$address/consist";
+
+
+    /**
+     * This implementation does not support advanced consists, so return true.
+     *
+     */
+    @Override
+    public boolean isCommandStationConsistPossible() {
+        return true;
+    }
+
+    /**
+     * Does a CS consist require a separate consist address? CS consist
+     * addresses are assigned by the user, so return true.
+     *
+     */
+    @Override
+    public boolean csConsistNeedsSeperateAddress() {
+        return true;
+    }
+
+    /**
+     * Add a new MQTT Consist with the given address to
+     * consistTable/consistList.
+     */
+    @Override
+    public Consist addConsist(LocoAddress address) {
+        if (! (address instanceof DccLocoAddress)) {
+            throw new IllegalArgumentException("address is not a DccLocoAddress object");
+        }
+        if (consistTable.containsKey(address)) { // no duplicates allowed across all connections
+            return consistTable.get(address);
+        }
+        MqttConsist consist;
+        consist = new MqttConsist((DccLocoAddress) address, (MqttSystemConnectionMemo) adapterMemo,
```

Comment:
Static analysis failes on this line due to the unnecessary cast of `adapterMemo` to `MqttSystemConnectionMemo`.

Reply:
Fixed